base/partr.jl

```diff
@@ -61,45 +43,76 @@ function sift_up(heap::taskheap, idx::Int32)
 end
 
 
-function sift_down(heap::taskheap, idx::Int32)
+function multiq_sift_down(heap::taskheap, idx::Int32)
     if idx <= heap.ntasks
         for child = (heap_d * idx - heap_d + Int32(2)):(heap_d * idx + Int32(1))
-            child > tasks_per_heap && break
+            child > length(heap.tasks) && break
             if isassigned(heap.tasks, Int(child)) &&
                     heap.tasks[child].priority < heap.tasks[idx].priority
                 t = heap.tasks[idx]
                 heap.tasks[idx] = heap.tasks[child]
                 heap.tasks[child] = t
-                sift_down(heap, child)
+                multiq_sift_down(heap, child)
             end
         end
     end
 end
 
 
+function multiq_size()
+    heap_c = UInt32(2)
+    heap_p = UInt32(length(heaps))
+    nt = UInt32(nthreads())
+
+    if heap_c * nt <= heap_p
+        return heap_p
+    end
+
+    @lock heaps_lock begin
+        heap_p = UInt32(length(heaps))
+        nt = UInt32(nthreads())
+        if heap_c * nt <= heap_p
+            return heap_p
+        end
+
+        heap_p += heap_c * nt
```

Comment:
This should not be `+=`. The lowest delay (and also the lowest rank error) for `deletemin` is with `heap_c = 2`; delay and error scale linearly with `heap_c`. We probably do not want `heap_c` to ever exceed 4.

Reply:
Why not just use `=` here?

Reply:
Doesn't scale as well if the thread count is changing dynamically. This initializes to 2x, then averages around 3x if the user is adding threads later.